### ISSUE TITLE
Japanese notation bugs

### ISSUE THREAD

**Issue by IcefoxKishi** [WandererXII/lishogi#874]
**Exact URL where the bug happened**
SFEN ```5l3/3S3S1/2b6/4GS3/2r1GK1G1/3G1S2G/9/3S3S1/5r3 b - 1``` in the analysis board and Japanese notation.
**Steps to reproduce the bug**
Play the moves 6b5c, 2b3c, 5d6d, 6e5e, 6h7g, 2h3g, 1f1e (promotion doesn't matter).
**What did you expect to happen?**
Those 7 moves would be recorded in Japanese notation as:
５三銀引（不）成、３三銀引（不）成、６四金寄、５六金寄、５七銀上、３七銀上、１五金上 in that order.
**What happened instead?**
The moves were displayed like this, with no movement indicators (except the 7th move which had a wrong movement indicator).
![image](https://github.com/user-attachments/assets/9e3dab83-7089-4753-a5bc-0c38fc0d75bd)
**Operating system and browser version**
Windows 11, Firefox 131.0

**Comment by IcefoxKishi** [comment]
IMO this is due to Lishogi straight up ruling out illegal moves hence the first 6 moves were displayed like so. The 7th move I have no idea why. 直 is only used when there are 2 pieces on the same row and one is moving straight up.

**Comment by WandererXII** [comment]
I will fix the 直 if 上 can be used. About the rest it's as you said, illegal moves are not considered as alternatives so no need to disambiguate (https://github.com/WandererXII/shogiops/blob/master/src/notation/util.ts#L6). But I will rewrite it to consider even illegal moves, if that's the correct behavior. Thanks for reporting this and providing an example.

**Comment by IcefoxKishi** [comment]
Yes, although you can't play illegal moves, they are still "possibilities" (even if they cost you your king).

**Comment by WandererXII** [comment]
Was fixed here (https://github.com/WandererXII/shogiops/commit/4713228576f1d3374d3d86b4af0f11f1f2f6b730), changes will be live with the next tournament update. Thanks!
